File: 2022/06/6.clj
(ns aoc.2022.06.6
  (:require [aoc.common :refer [read-input]]))

(defn find-marker [signal distinct]
  (->> (range (count signal))
       (filter #(-> signal (subvec % (+ % distinct)) set count (= distinct)))
       first
       (+ distinct)))

(let [signal (read-input :split-with #"")]
  (println "Part 1:" (find-marker signal 4))
  (println "Part 2:" (find-marker signal 14)))

File: 2022/06/6.py
signal = open("input.txt").read()

def find_marker(signal, distinct):
  for i in range(len(signal)):
    if len(set(signal[i:i + distinct])) == distinct:
      return i + distinct

print("Part 1:", find_marker(signal, 4))
print("Part 2:", find_marker(signal, 14))